### ISSUE TITLE
Question about Model parallelism in Keras

### ISSUE THREAD

**Issue by madhurinagare** [keras-team/keras#14195]
I am trying to implement model parallelism in Keras.

I am using Keras-2.2.4
Tensorflor-1.13.1 

Rough structure of my code is : 
import tensorflow as tf
import keras

**def model_definition():**
    input0 =  Input(shape = (None, None)) 
    input1 = Input(shape = (None, None))
    with tf.Session(config=tf.ConfigProto(allow_soft_placement=False, log_device_placement=True)):
        model = get_some_CNN_model()
        with tf.device(tf.DeviceSpec(device_type="GPU", device_index=0)):
            op0 = model(input0)
        with tf.device(tf.DeviceSpec(device_type="GPU", device_index=1)):
            op1 = model(input1)
        with tf.device(tf.DeviceSpec(device_type="CPU", device_index=0)):
            concatenated_ops = concatenate([op0, op1],axis=-1,  name = 'check_conc1')
    mixmodel = Model(inputs=[input0, input1], outputs =  concatenated_ops)
    **return mixmodel**

mymodel = model_definition()
mymodel.fit_generator()

Expected result: While training, computations for op0 and op1 should be done on gpu0 and gpu1, respectively.

Problem 1: When I have 2 gpus available, the training works fine. nvidia-smi shows that both gpus are being used. Though I am not sure if both gpus are doing their intended work. How to confirm that?
Because, even I set log_device_placement as True, I don't see any task allocated to gpu 1

Problem 2: When I run this code on a machine with 1 GPU available, still it runs fine. It is expected show an error because GPU 1 is not available.

The example shown [here](https://stackoverflow.com/questions/42069147/implementation-of-model-parallelism-in-tensorflow) works fine as expected. It doesn't show the problem 2, i.e. on single GPU, it raises an error.
 
So I think some manipulation is happening inside keras.

I have also tried using import tensorflow.python.keras instead of import keras, in case it is causing any conflict.
However, the both problems persist.

Would appreciate any clue about this issue. Thank you.

**Comment by tilakrayal** [comment]
@madhurinagare ,
Could you please update tf and keras to the latest stable version v.2.6 and let us know if you are facing the same error. Thanks!
